ethereum-client/src/lib.rs

```diff
@@ -126,96 +199,227 @@ fn send_rpc(
         .map_err(|_| EthereumClientError::HttpTimeout)?;
 
     if response.code != 200 {
-        debug::warn!("Unexpected status code: {}", response.code);
+        log!("Unexpected status code: {}", response.code);
         return Err(EthereumClientError::HttpErrorCode(response.code));
     }
 
     let body = response.body().collect::<Vec<u8>>();
 
     // Create a str slice from the body.
     let body_str = sp_std::str::from_utf8(&body).map_err(|_| {
-        debug::warn!("No UTF8 body");
+        log!("No UTF8 body");
         EthereumClientError::InvalidUTF8
     })?;
+    debug!("RPC Response: {}", body_str.clone());
 
     Ok(String::from(body_str))
 }
 
-pub fn fetch_and_decode_logs(
+pub fn get_block(
     server: &str,
-    params: Vec<&str>,
-) -> Result<Vec<EthereumLogEvent>, EthereumClientError> {
-    let body_str: String = send_rpc(server, "eth_getLogs", params)?;
-    let deserialized_body = deserialize_get_logs_response(&body_str)
-        .map_err(|_| EthereumClientError::JsonParseError)?;
-    let eth_logs = deserialized_body
+    eth_starport_address: &[u8; 20],
+    block_num: EthereumBlockNumber,
+) -> Result<EthereumBlock, EthereumClientError> {
+    let block_str = encode_block_hex(block_num);
+    let block_obj = get_block_object(server, &block_str)?;
+    debug!("eth_starport_address: {:X?}", &eth_starport_address[..]);
+    let get_logs_params = vec![serde_json::json!({
+        "address": format!("0x{}", &eth_starport_address[..].iter().map(|x| format!("{:02x}", x)).collect::<String>()),
```

Comment:
We should remove this hack and just use `::hex::encode` now that we don't do WASM replacements.